Sources/Utility/ArgumentParser.swift

```diff
@@ -429,13 +437,17 @@ public final class ArgumentParser: ArgumentParserProtocol {
                 }
                 try result.addResult(for: positionalArg, result: resultValue)
             } else {
+                let (arg, value) = arg.split(around: "=")
+                associatedArgumentValue = value
                 // Get the corresponding option for the option argument.
                 guard let option = optionsMap[arg] else {
                     throw ArgumentParserError.unknown(option: arg)
                 }
                 // Initialize the argument and add to result.
                 let resultValue = try option.kind.init(parser: self)
```

Comment:
I would probably feel more comfortable if instead of using mutable state, this used a tiny little wrapper struct to pass down the protocol which included the associated value in it, if that makes sense. This makes it easier to reason about, and makes it obvious exactly which parts of the API of this parser are being exposed (without needing to reference the protocol).

This will force you to deal with what it means if a kind implementation does `next() next() next()`and there is an associated value, which is a good thing.